src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSBuildReferenceTaxonomyUtils.java

```diff
@@ -0,0 +1,318 @@
+     * taxonomic ID, it instead gets added to taxIdToProperties. Later we merge both results into taxIdToProperties.
+     * Method: First, look for either "taxid|<taxid>|" or "ref|<accession>|" in the sequence name. If neither of
+     * those are found, use the first word of the name as the accession.
+     */
+    protected static Map<String, Tuple2<String, Long>> parseReferenceRecords(final List<SAMSequenceRecord> dictionaryList,
+                                                                             final Map<String, PSPathogenReferenceTaxonProperties> taxIdToProperties) {
+
+        final Map<String, Tuple2<String, Long>> accessionToNameAndLength = new HashMap<>();
+        for (final SAMSequenceRecord record : dictionaryList) {
+            final String recordName = record.getSequenceName();
+            final long recordLength = record.getSequenceLength();
+            final String[] tokens = recordName.split("[|]");
+            String recordAccession = null;
+            String recordTaxId = null;
+            for (int i = 0; i < tokens.length - 1 && recordTaxId == null; i++) {
+                if (tokens[i].equals("ref")) {
+                    recordAccession = tokens[i + 1];
+                } else if (tokens[i].equals("taxid")) {
+                    recordTaxId = tokens[i + 1];
+                }
+            }
+            if (recordTaxId == null) {
+                if (recordAccession == null) {
+                    final String[] tokens2 = tokens[0].split(" "); //Default accession to first word in the name
+                    recordAccession = tokens2[0];
+                }
+                accessionToNameAndLength.put(recordAccession, new Tuple2<>(recordName, recordLength));
+            } else {
+                updateAccessionTaxonProperties(recordTaxId, recordName, recordLength, taxIdToProperties);
+            }
+        }
+        return accessionToNameAndLength;
+    }
+
+    /**
+     * Builds maps of reference contig accessions to their taxonomic ids and vice versa.
+     * Input can be a RefSeq or Genbank catalog file. accNotFound is an initial list of
+     * accessions from the reference that have not been successfully looked up; if null,
+     * will be initialized to the accToRefInfo key set by default.
+     * <p>
+     * Returns a collection of reference accessions that could not be found, if any.
+     */
+    protected static Set<String> parseCatalog(final BufferedReader reader,
+                                              final Map<String, Tuple2<String, Long>> accessionToNameAndLength,
+                                              final Map<String, PSPathogenReferenceTaxonProperties> taxIdToProperties,
+                                              final boolean bGenBank, Set<String> accessionsNotFound) {
+        try {
+            String line;
+            final int taxIdColumnIndex, accessionColumnIndex;
+            if (bGenBank) {
+                taxIdColumnIndex = 6;
```

Comment:
Added an interface AccessionCatalogFormat with getTaxIdColumn() and getAccessionColumn() methods. Classes RefSeqCatalogFormat and GenBankCatalogFormat implement this interface for the two types of catalogs.